pkgs/by-name/li/libvibrant/package.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, makeWrapper
+, libX11
+, libXrandr
+, linuxPackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libvibrant";
+  version = "2100c09";
+
+  src = fetchFromGitHub {
+    owner = "libvibrant";
+    repo = pname;
```

Comment:
```suggestion
    repo = "libvibrant";
```

Reply:
Why repeat the name of it if it's the same as the package name?

Reply:
Because, in short words, Nix is not a macro language.

There is no a priori connection between `pname` and `src.repo`. They are the same here by a convention established by the upstream devteam, not by something else enforcing this.
Indeed, many other softwares do not follow this pattern. 
What about openmsx, where `src.repo == OpenMSX`. What should we do on this case? A `pname = lowercase(src.repo);`?
What about PIL, the defunct Python Imaging Library? Its source code was stored on a tarball with name `Imaging-${version}.tgz`. I can't imagine how to transform from pil to Imaging...

As an exercise, just try to change each and every occurrence of `libvibrant` to `pname`. What is the gain besides a mess of string interpolations?

Also, it creates a chain of dependencies between lines of code that otherwise would work fine without that interleaved dependency. Such dependencies make a little bit harder to reason statically about the code.

Reply:
understood, thanks for the detailed explanation